--- a/mt_wifi/embedded/sta/sanity.c
+++ b/mt_wifi/embedded/sta/sanity.c
@@ -391,7 +391,7 @@ BOOLEAN PeerAssocRspSanity(
 #endif /* DOT11R_FT_SUPPORT */
 			/* Copy whole RSNXE context */
 			if ((pEid->Len + 2) <= sizeof(ie_list->rsnxe_ie))
-				NdisMoveMemory(&ie_list->rsnxe_ie_len, pEid, pEid->Len + 2);
+				NdisMoveMemory(&ie_list->rsnxe_ie, pEid, pEid->Len + 2);
 			else
 				MTWF_DBG(NULL, DBG_CAT_CLIENT, DBG_SUBCAT_ALL, DBG_LVL_ERROR,
 					"No enough buffer size for ie_list->rsnxe_ie!\n");
