--- a/mt_wifi/embedded/common/qos.c
+++ b/mt_wifi/embedded/common/qos.c
@@ -541,7 +541,7 @@
 }
 
 static void wext_send_mscs_classifier_parameter(struct wifi_dev *wdev,
-	UCHAR *classifier_parameter_report, ULONG len)
+	struct classifier_parameter *classifier_parameter_report)
 {
 	char *buf = NULL;
 	struct wapp_event *event = NULL;
@@ -570,7 +570,7 @@
 	*/
 	req_data = (struct classifier_parameter *)&(event->data);
 
-	NdisCopyMemory(req_data, classifier_parameter_report, sizeof(*req_data));
+	*req_data = *classifier_parameter_report;
 	RtmpOSWrielessEventSend(wdev->if_dev, RT_WLAN_EVENT_CUSTOM,
 		OID_WAPP_EVENT, NULL, (PUCHAR)buf, buflen);
 	os_free_mem(buf);
@@ -609,7 +609,7 @@
 	ptr += 7;
 	leftlen -= 7;
 	if (cs_param.requet_type == SCS_REQ_TYPE_REMOVE) {
-		wext_send_mscs_classifier_parameter(wdev, (UCHAR *)&cs_param, sizeof(cs_param));
+		wext_send_mscs_classifier_parameter(wdev, &cs_param);
 		return 0;
 	}
 	while ((*ptr) == IE_WLAN_EXTENSION && (*(ptr + 2)) == IE_EXTENSION_ID_TCLAS_MASK) {
@@ -621,8 +621,7 @@
 		if ((*(ptr+1)) == MSCS_CLASSIFIER_TYPE4) {
 			cs_param.cs.header.cs_type = *(ptr + 1);
 			cs_param.cs.header.cs_mask = *(ptr + 2);
-			wext_send_mscs_classifier_parameter(wdev,
-				(UCHAR *)&cs_param, sizeof(cs_param));
+			wext_send_mscs_classifier_parameter(wdev, &cs_param);
 		} else
 			MTWF_DBG(NULL, DBG_CAT_PROTO, DBG_SUBCAT_ALL, DBG_LVL_ERROR,
 				"%s, not support mscs classifier type= %d.\n", __func__, (*ptr));
