--- a/package/firmware/linux-firmware/mediatek.mk
+++ b/package/firmware/linux-firmware/mediatek.mk
@@ -78,25 +78,6 @@
 endef
 $(eval $(call BuildPackage,mt7925bt-firmware))
 
-Package/mt7981-wo-firmware = $(call Package/firmware-default,MT7981 offload firmware,,LICENCE.mediatek)
-define Package/mt7981-wo-firmware/install
-	$(INSTALL_DIR) $(1)/lib/firmware/mediatek
-	$(INSTALL_DATA) \
-		$(PKG_BUILD_DIR)/mediatek/mt7981_wo.bin \
-		$(1)/lib/firmware/mediatek
-endef
-$(eval $(call BuildPackage,mt7981-wo-firmware))
-
-Package/mt7986-wo-firmware = $(call Package/firmware-default,MT7986 offload firmware,,LICENCE.mediatek)
-define Package/mt7986-wo-firmware/install
-	$(INSTALL_DIR) $(1)/lib/firmware/mediatek
-	$(INSTALL_DATA) \
-		$(PKG_BUILD_DIR)/mediatek/mt7986_wo_0.bin \
-		$(PKG_BUILD_DIR)/mediatek/mt7986_wo_1.bin \
-		$(1)/lib/firmware/mediatek
-endef
-$(eval $(call BuildPackage,mt7986-wo-firmware))
-
 Package/mt7987-2p5g-phy-firmware = $(call Package/firmware-default,MT7987 built-in 2.5G Ethernet PHY firmware,,LICENCE.mediatek)
 define Package/mt7987-2p5g-phy-firmware/install
 	$(INSTALL_DIR) $(1)/lib/firmware/mediatek/mt7987
