diff --git a/package/network/config/netifd/files/lib/netifd/utils.uc b/package/network/config/netifd/files/lib/netifd/utils.uc
index 84db69d..11b4906 100644
--- a/package/network/config/netifd/files/lib/netifd/utils.uc
+++ b/package/network/config/netifd/files/lib/netifd/utils.uc
@@ -54,8 +54,12 @@ export function handler_load(path, cb)
 		system(`./${script} "" "dump" >&${f.fileno()}`);
 		chdir(prev_dir);
 		f.seek();
-		while (!f.error()) {
-			let data = trim(f.read("line"));
+		while (true) {
+			let line = f.read("line");
+			if (!line)
+				break;
+
+			let data = trim(line);
 			try {
 				data = json(data);
 			} catch (e) {
