--- a/warp_proc.c
+++ b/warp_proc.c
@@ -173,7 +173,7 @@ wdma_proc_show(struct seq_file *seq, voi
 static int
 wdma_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, wdma_proc_show, PDE_DATA(file_inode(file)));
+	return single_open(file, wdma_proc_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -187,7 +187,7 @@ wdma_proc_write(struct file *file, const
 	char value[64] = "";
 	char *end = NULL;
 	char choice;
-	struct wdma_entry *wdma = (struct wdma_entry *)PDE_DATA(file_inode(file));
+	struct wdma_entry *wdma = (struct wdma_entry *)pde_data(file_inode(file));
 
 	if (buff && !copy_from_user(value, buff, len1)) {
 		choice = warp_str_tol(value, &end, 10);
@@ -263,7 +263,7 @@ warp_proc_handle(struct seq_file *seq, s
 static int
 wed_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, wed_proc_show, PDE_DATA(file_inode(file)));
+	return single_open(file, wed_proc_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -276,7 +276,7 @@ wed_proc_write(struct file *file, const
 	char value[64] = "";
 	char *end = NULL;
 	char choice;
-	struct wed_entry *wed = (struct wed_entry *)PDE_DATA(file_inode(file));
+	struct wed_entry *wed = (struct wed_entry *)pde_data(file_inode(file));
 
 	if (buff && !copy_from_user(value, buff, len1)) {
 		choice = warp_str_tol(value, &end, 10);
@@ -320,7 +320,7 @@ warp_proc_cr_show(struct seq_file *seq,
 static int
 warp_proc_cr_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_cr_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_cr_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -339,7 +339,7 @@ warp_proc_cr_write(struct file *file, co
 	char *s = value;
 	unsigned long addr;
 	u32 cr_value = 0;
-	struct warp_entry *warp = (struct warp_entry *)PDE_DATA(file_inode(file));
+	struct warp_entry *warp = (struct warp_entry *)pde_data(file_inode(file));
 
 	if (buff && !copy_from_user(value, buff, len1)) {
 		token = strsep(&s, " ");
@@ -433,7 +433,7 @@ warp_proc_ctrl_show(struct seq_file *seq
 static int
 warp_proc_ctrl_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_ctrl_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_ctrl_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -450,7 +450,7 @@ warp_proc_ctrl_write(struct file *file,
 	char is_true;
 	char type;
 	char *s = value;
-	struct warp_entry *warp = (struct warp_entry *)PDE_DATA(file_inode(file));
+	struct warp_entry *warp = (struct warp_entry *)pde_data(file_inode(file));
 
 	if (len1 > 64 || len1 == 0) {
 		warp_dbg(WARP_DBG_OFF, "%s(): invalid length len1 %d!\n", __func__, len1);
@@ -509,7 +509,7 @@ warp_proc_trace_show(struct seq_file *se
 static int
 warp_proc_trace_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_trace_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_trace_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -526,7 +526,7 @@ warp_proc_trace_write(struct file *file,
 	char *token;
 	char *s = value;
 	unsigned int tmp;
-	struct warp_ctrl *ctrl = (struct warp_ctrl *)PDE_DATA(file_inode(file));
+	struct warp_ctrl *ctrl = (struct warp_ctrl *)pde_data(file_inode(file));
 	struct warp_cputracer *tracer = &ctrl->bus.tracer;
 
 	if (buff && !copy_from_user(value, buff, len1)) {
@@ -630,7 +630,7 @@ warp_proc_conf_show(struct seq_file *seq
 static int
 warp_proc_conf_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_conf_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_conf_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -647,7 +647,7 @@ warp_proc_conf_write(struct file *file,
 	char *token;
 	char *s = value;
 	unsigned int tmp;
-	struct warp_ctrl *ctrl = (struct warp_ctrl *)PDE_DATA(file_inode(file));
+	struct warp_ctrl *ctrl = (struct warp_ctrl *)pde_data(file_inode(file));
 
 	if (buff && !copy_from_user(value, buff, len1)) {
 		token = strsep(&s, ":");
@@ -798,7 +798,7 @@ warp_proc_state_show(struct seq_file *se
 static int
 warp_proc_state_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_state_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_state_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -812,7 +812,7 @@ warp_proc_state_write(struct file *file,
 	char value[64];
 	char *end;
 	char log;
-	struct warp_entry *warp = (struct warp_entry *)PDE_DATA(file_inode(file));
+	struct warp_entry *warp = (struct warp_entry *)pde_data(file_inode(file));
 
 	if (buff && !copy_from_user(value, buff, len1)) {
 		log = warp_str_tol(value, &end, 10);
@@ -850,7 +850,7 @@ warp_proc_cfg_show(struct seq_file *seq,
 static int
 warp_proc_cfg_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_cfg_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_cfg_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -871,7 +871,7 @@ warp_proc_tx_show(struct seq_file *seq,
 static int
 warp_proc_tx_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_tx_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_tx_show, pde_data(file_inode(file)));
 }
 
 /*
@@ -892,7 +892,7 @@ warp_proc_rx_show(struct seq_file *seq,
 static int
 warp_proc_rx_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_rx_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_rx_show, pde_data(file_inode(file)));
 }
 
 
@@ -932,7 +932,7 @@ warp_proc_rxbm_show(struct seq_file *seq
 static int
 rxbm_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, warp_proc_rxbm_show, PDE_DATA(file_inode(file)));
+	return single_open(file, warp_proc_rxbm_show, pde_data(file_inode(file)));
 }
 
 
@@ -966,7 +966,7 @@ wo_ee_proc_show(struct seq_file *seq, vo
 static int
 wo_ee_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, wo_ee_proc_show, PDE_DATA(file_inode(file)));
+	return single_open(file, wo_ee_proc_show, pde_data(file_inode(file)));
 }
 
 
@@ -1030,14 +1030,14 @@ wo_proc_show(struct seq_file *seq, void
 static int
 wo_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, wo_proc_show, PDE_DATA(file_inode(file)));
+	return single_open(file, wo_proc_show, pde_data(file_inode(file)));
 }
 
 static ssize_t
 wo_proc_write(struct file *file, const char __user *usr_buff, size_t len1, loff_t *ppos)
 {
 	u8 *buffer = NULL, *type = NULL, *sub_str = NULL, *input[11] = {0};
-	struct warp_entry *warp = (struct warp_entry *)PDE_DATA(file_inode(file));
+	struct warp_entry *warp = (struct warp_entry *)pde_data(file_inode(file));
 	int scan_num = 0, input_idx = 0, input_total = 0;
 	struct warp_msg_cmd cmd = {0};
 	u8 sendmsg_buf[128] = {0};
@@ -1172,118 +1172,105 @@ error:
 /*
  * global file operation
 */
-static const struct file_operations proc_warp_trace_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_trace_open,
-	.write  = warp_proc_trace_write,
-	.read	= seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_trace_fops = {
+	.proc_open = warp_proc_trace_open,
+	.proc_write  = warp_proc_trace_write,
+	.proc_read	= seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_warp_conf_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_conf_open,
-	.write  = warp_proc_conf_write,
-	.read	= seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_conf_fops = {
+	.proc_open = warp_proc_conf_open,
+	.proc_write  = warp_proc_conf_write,
+	.proc_read	= seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_warp_cr_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_cr_open,
-	.write  = warp_proc_cr_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_cr_fops = {
+	.proc_open = warp_proc_cr_open,
+	.proc_write  = warp_proc_cr_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_warp_stat_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_state_open,
-	.write  = warp_proc_state_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_stat_fops = {
+	.proc_open = warp_proc_state_open,
+	.proc_write  = warp_proc_state_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_warp_cfg_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_cfg_open,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_cfg_fops = {
+	.proc_open = warp_proc_cfg_open,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_warp_tx_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_tx_open,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_tx_fops = {
+	.proc_open = warp_proc_tx_open,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_warp_rx_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_rx_open,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_warp_rx_fops = {
+	.proc_open = warp_proc_rx_open,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_wed_fops = {
-	.owner = THIS_MODULE,
-	.open = wed_proc_open,
-	.write  = wed_proc_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_wed_fops = {
+	.proc_open = wed_proc_open,
+	.proc_write  = wed_proc_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_wdma_fops = {
-	.owner = THIS_MODULE,
-	.open = wdma_proc_open,
-	.write  = wdma_proc_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_wdma_fops = {
+	.proc_open = wdma_proc_open,
+	.proc_write  = wdma_proc_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_ctrl_fops = {
-	.owner = THIS_MODULE,
-	.open = warp_proc_ctrl_open,
-	.write  = warp_proc_ctrl_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_ctrl_fops = {
+	.proc_open = warp_proc_ctrl_open,
+	.proc_write  = warp_proc_ctrl_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
 
-static const struct file_operations proc_rxbm_fops = {
-	.owner = THIS_MODULE,
-	.open = rxbm_proc_open,
-	.write  = rxbm_proc_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_rxbm_fops = {
+	.proc_open = rxbm_proc_open,
+	.proc_write  = rxbm_proc_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_wo_fops = {
-	.owner = THIS_MODULE,
-	.open = wo_proc_open,
-	.write  = wo_proc_write,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_wo_fops = {
+	.proc_open = wo_proc_open,
+	.proc_write  = wo_proc_write,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
-static const struct file_operations proc_woee_fops = {
-	.owner = THIS_MODULE,
-	.open = wo_ee_proc_open,
-	.read  = seq_read,
-	.llseek = seq_lseek,
-	.release = single_release,
+static const struct proc_ops proc_woee_fops = {
+	.proc_open = wo_ee_proc_open,
+	.proc_read  = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
 };
 
 
